python/tskit/trees.py

```diff
@@ -2116,15 +2116,45 @@ def num_tracked_samples(self, u=None):
         u = self.virtual_root if u is None else u
         return self._ll_tree.get_num_tracked_samples(u)
 
-    # TODO document these traversal arrays
-    # https://github.com/tskit-dev/tskit/issues/1788
     def preorder(self, u=NULL):
+        """
+        Returns a numpy array of node ids. Starting at `u`, yield the current node,
+        then recurse and do a preorder on each child of the current node. See also
+        `Wikipedia<https://en.wikipedia.org/wiki/Tree_traversal#Pre-order_(NLR)>`__.
+        Also see :ref:`tutorials:sec_analysing_trees_traversals` for examples of how
+        to use traversals.
+
+        :param int u: The node to start from, defaults to the virtual root.
```

Comment:
This isn't quite right. If we start from the virtual root we return the virtual root in the arrays, but this isn't what we usually want. It's more like "if specified, return all nodes in the subtree rooted at u (including u) in traversal order".

Reply:
Fixed